Sprint-3/todo-list/script.js

```diff
@@ -12,14 +45,92 @@ let todos = [
 
 populateTodoList(todos);
 
+function toggleComplete(index) {
+  // function to toggle complete status
+  todos[index].completed = !todos[index].completed;
+  populateTodoList(todos);
+}
+
+function deleteTodo(index) {
+  //function to delete specific todo
+  todos.splice(index, 1);
+  populateTodoList(todos);
+}
+populateTodoList(todos); 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+    let input =
+    document.getElementById("new-todo-input") ||
+    document.getElementById("todo-input") ||
+    document.querySelector('input[type="text"]');
+
+  let todoText = input.value.trim();
+
+  if (todoText === "") {
+    alert("Please enter a todo item!");
+    return;
+  }
+  let newTodo = {
+    // create new todo object
+    task: todoText,
+    completed: false,
+  };
+
+  todos.push(newTodo); // ad todos array
+
+  input.value = ""; //clear input field
+
+  populateTodoList(todos); //refresh the todo list
 }
 
+
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
+  todos = todos.filter((todo) => !todo.completed); // filter out the completed todos
+
+  populateTodoList(todos); // refresh the display
+
+  console.log("The completed todos are deleted!");
 }
+// Event listener
+document.addEventListener("DOMContentLoaded", function () {
+  // Try to find the form by different selectors
+  let form =
+    document.querySelector("form") ||
+    document.querySelector(".add-todo-form") ||
+    document.getElementById("todo-form");
+
+  if (form) {
+    form.addEventListener("submit", addNewTodo);
+  } else {
+    console.log(
+      "No form found. Make sure your HTML has a form element or call addNewTodo() directly from a button onclick."
+    );
+
+    // Try to find an add button and attach the event
+    let addButton =
+      document.querySelector('button[type="submit"]') ||
+      document.querySelector(".add-btn") ||
+      document.getElementById("add-todo-btn");
+
+    if (addButton) {
+      addButton.addEventListener("click", addNewTodo);
+    }
+  }
+
+  // Try to find delete all completed button
+  let deleteAllBtn =
```

Comment:
If there isn't an item in the list, can you think of a way to disable the 'Delete All Complete' button. Ideally, the user shouldn't be able to click on the button if there aren't any items to delete!